sw/host/opentitantool/src/command/lc.rs

```diff
@@ -136,6 +136,57 @@ impl CommandDispatch for LcRegRead {
     }
 }
 
+#[derive(Debug, Args)]
+/// Reads the 256bit device ID over JTAG.
+pub struct LcDeviceIdRead {
+    /// Reset duration when switching the LC TAP straps.
+    #[arg(long, value_parser = parse_duration, default_value = "100ms")]
+    pub reset_delay: Duration,
+
+    #[command(flatten)]
+    pub jtag_params: JtagParams,
+}
+
+impl CommandDispatch for LcDeviceIdRead {
+    fn run(
+        &self,
+        _context: &dyn Any,
+        transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Annotate>>> {
+        // Set the TAP straps for the lifecycle controller and reset.
+        transport.pin_strapping("PINMUX_TAP_LC")?.apply()?;
+        transport.reset_target(self.reset_delay, true)?;
+
+        // Spawn an OpenOCD process and connect to the LC JTAG TAP.
+        let mut jtag = self
+            .jtag_params
+            .create(transport)?
+            .connect(JtagTap::LcTap)?;
+
+        // Read and concatenate device ID registers.
+        let offsets = [
+            LcCtrlReg::DeviceId0,
+            LcCtrlReg::DeviceId1,
+            LcCtrlReg::DeviceId2,
+            LcCtrlReg::DeviceId3,
+            LcCtrlReg::DeviceId4,
+            LcCtrlReg::DeviceId5,
+            LcCtrlReg::DeviceId6,
+            LcCtrlReg::DeviceId7,
+        ];
+
+        let mut value = String::from("");
+        for off in offsets {
+            value = format!("{:08x}", jtag.read_lc_ctrl_reg(&off)?) + &value;
+        }
```

Comment:
This might be better as:
```
let mut value = vec![String::from("0x")];
for off in offsets {
    value.push(format!("{:08x}", jtag.read_lc_ctrl_reg(&off)?));
}

jtag.disconnect()?;
Ok(Some(Box::new(LcRegReadResult {
    value: value.join(""),
})))
```

Reply:
Thanks, changed!

I also reversed the order of the 32bit words, since we need: `"0x" + str(word7) + str(word6) + ... + str(word0)`